### PR TITLE
fix: Disable Delete and Save buttons in DeleteDialog

### LINKED ISSUE #1338: [BUG][FormTribe 🔥][100 Points] Delete and Save Buttons Remain Clickable After Click on Delete or Save button
### Issue Summary

Upon clicking the "Delete" button in the DeleteDialog component, the button does not become disabled as expected. Consequently, users can click the "Delete" button multiple times, leading to potential issues. The same behavior is observed with the "Save" button. 

Multiple time click create error also 

image

### Steps to Reproduce

1. Open the DeleteDialog component.
2. Click the "Delete" button.
3. Observe that the button does not become disabled.
4. Attempt to click the "Delete" button again. Note that it remains clickable.
5. Repeat the above steps for the "Save" button.
image

https://www.loom.com/share/9b9980e2125b443589acd6c0a11f55e4?sid=71ec4b52-033b-4674-bfff-8a20330f7437

### Expected behavior

After clicking the "Delete" button, it should be disabled to prevent multiple consecutive clicks until the action is completed.
The "Save" button should exhibit the same behavior, becoming disabled temporarily after being clicked to avoid multiple rapid clicks.

https://www.loom.com/share/c4973ca8c83a4f289e8b63b0a8c9dad2?sid=aa956af4-9871-4e10-bd16-370ef253f6d9

### Other information

_No response_

### Screenshots

image

### Environment

- [X] Formbricks Cloud (app.formbricks.com)
- [X] Self-hosted Formbricks

### Desktop (please complete the following information)

```markdown
- OS: Windows
- Node: v18.18.0
- npm: 9.8.1
```

### PR DESCRIPTION
**Pull request by shyamraghu**
## What does this PR do?

This PR addresses the issue where the "Delete" and "Save" buttons in the DeleteDialog component were not being disabled after being clicked, allowing multiple rapid clicks and potentially causing unintended actions.

Fixes #1338

 Loom Video: https://www.loom.com/share/c4973ca8c83a4f289e8b63b0a8c9dad2?sid=b52b9be8-bc6c-40dd-8b2e-b7ed910fa9e9

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Open the DeleteDialog component.
- Click the "Delete" button.
- Verify that the "Delete" button becomes disabled after the first click.
- Attempt to click the "Delete" button again to ensure it remains disabled.
- Repeat the above steps for the "Save" button.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [] Ran `pnpm build`
- [x] Checked for warnings; there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by jobenjada** [comment]
Hey @shyamraghu  !

Please don't disable the button but add a loading state. The Button component comes with a loading state already :)

**Comment by shyamraghu** [comment]
> Hey @shyamraghu  !
> 
> Please don't disable the button but add a loading state. The Button component comes with a loading state already :)

Please assign it to me

File: packages/ui/DeleteDialog/index.tsx [hunk 1]
```diff
@@ -43,7 +46,7 @@ export function DeleteDialog({
           }}>
           {useSaveInsteadOfCancel ? "Save" : "Cancel"}
         </Button>
-        <Button variant="warn" onClick={onDelete} loading={isDeleting} disabled={disabled}>
```
Comment (jobenjada) [inline packages/ui/DeleteDialog/index.tsx:46]: why did you remove the disabled flag? Since this component is use in several locations removing it might break functionality somewhere else. Did you run a build locally to confirm that it doesnt?

**Review by jobenjada — CHANGES_REQUESTED** [review]

**Comment by jobenjada** [comment]
@shyamraghu its yours, your on it ;) 

Did you have a look at the Publish button or any other buttons? The Publish button does have a loading state but at least locally, it seems to break. Often I have to hit the button twice. Please have a look to see if you can spot some UX inconsistencies 🙏

**Comment by shyamraghu** [comment]
> @shyamraghu its yours, your on it ;)
> 
> Did you have a look at the Publish button or any other buttons? The Publish button does have a loading state but at least locally, it seems to break. Often I have to hit the button twice. Please have a look to see if you can spot some UX inconsistencies 🙏

The disabled prop value was undefined

File: packages/ui/DeleteDialog/index.tsx (hunk 1 above)
Comment (shyamraghu) [reply packages/ui/DeleteDialog/index.tsx:46]: The disabled prop value was undefined

**Comment by shyamraghu** [comment]
> @shyamraghu its yours, your on it ;)
> 
> Did you have a look at the Publish button or any other buttons? The Publish button does have a loading state but at least locally, it seems to break. Often I have to hit the button twice. Please have a look to see if you can spot some UX inconsistencies 🙏

Ok, I will look at that issue. Would you like to explane me some more about this issue or give me some screenshot or something.

**Comment by jobenjada** [comment]
This is a duplicate to the one with the loading state right? Closing it but let me know if I got that wrong
